src/shogun/features/DenseSubSamplesFeatures.h

```diff
@@ -174,6 +174,25 @@ template<class ST> class CDenseSubSamplesFeatures: public CDotFeatures
 	 * @param iterator as returned by get_feature_iterator
 	 */
 	virtual void free_feature_iterator(void* iterator);
+
+
+	/** does this class support compatible computation bewteen difference classes?
+	 * for example, this->dot(rhs_prt),
+	 * can rhs_prt be an instance of a difference class?
+	 *
+	 * @return whether this class supports compatible computation
+	 */
+	virtual bool support_compatible_class() const {return true;}
+
+	/** Given a class in right hand side, does this class support compatible computation? 
+	 *
+	 * for example, is this->dot(rhs_prt) valid,
+	 * where rhs_prt is the class in right hand side 
+	 *
+	 * @param rhs, the class in right hand side 
+	 * @return whether this class supports compatible computation
+	 */
+	virtual bool get_feature_class_compatibility (EFeatureClass rhs) const;
```

Comment:
this implementation also offer one way to deal with the issue https://github.com/shogun-toolbox/shogun/issues/2859